### PR TITLE
implement otel to services

### PR DESCRIPTION
**Pull request by jkralik**

### CODE REVIEW

File: pkg/opentelemetry/collector/client/grpc/client.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.10.0"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type Client struct {
+	client         *client.Client
+	tracerProvider *sdktrace.TracerProvider
+	closeFunc      []func()
```
Comment (Danielius1922) [inline pkg/opentelemetry/collector/client/grpc/client.go]: pkg/fn.FuncList?

**Review by Danielius1922 — APPROVED** [review]

File: pkg/opentelemetry/collector/client/grpc/client.go (hunk 1 above)
Comment (jkralik) [reply pkg/opentelemetry/collector/client/grpc/client.go]: fixed

**Review by ondrejtomcik — APPROVED** [review]

File: pkg/opentelemetry/collector/client/config.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+type GRPCConfig struct {
+	Enabled    bool          `yaml:"enabled"`
+	Connection client.Config `yaml:",inline"`
+}
+
+func (c *GRPCConfig) Validate() error {
+	if !c.Enabled {
+		return nil
+	}
+	if err := c.Connection.Validate(); err != nil {
```
Comment (Danielius1922) [inline pkg/opentelemetry/collector/client/config.go]: Lines 18-21 can be shortened to:

```
return c.Connection.Validate()
```

**Review by Danielius1922 — APPROVED** [review]

**Review by ondrejtomcik — APPROVED** [review]

File: pkg/opentelemetry/collector/client/config.go (hunk 2 above)
Comment (jkralik) [reply pkg/opentelemetry/collector/client/config.go]: Fixed

**Review by Danielius1922 — APPROVED** [review]

**Comment by Danielius1922** [comment]
Golangsci-lint and SonarCloud have some complains.

**Review by Danielius1922 — APPROVED** [review]

**Review by Danielius1922 — APPROVED** [review]

**Pull request merged**